scipy/interpolate/tests/test_rgi.py

```diff
@@ -484,6 +482,71 @@ def test_fill_value(self, method):
                                          method=method, bounds_error=False)
         assert np.isnan(interp([10]))
 
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values(self, method):
+        # Verify that non-scalar valued values also works
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5)] * 2 + [
+            (0.0, 5.0, 10.0, 15.0, 20, 25.0)
+        ] * 2
+
+        rng = np.random.default_rng(1234)
+        values = rng.random((6, 6, 6, 6, 8))
+        sample = rng.random((7, 11, 4))
+
+        interp = RegularGridInterpolator(points, values, method=method,
+                                         bounds_error=False)
+        v = interp(sample)
+        assert_equal(v.shape, (7, 11, 8), err_msg=method)
+
+        vs = []
+        for j in range(8):
+            interp = RegularGridInterpolator(points, values[..., j],
+                                             method=method,
+                                             bounds_error=False)
+            vs.append(interp(sample))
+        v2 = np.array(vs).transpose(1, 2, 0)
+
+        assert_allclose(v, v2, err_msg=method)
+
+    @parametrize_rgi_interp_methods
+    @pytest.mark.parametrize("num_trailing_dims", range(1, 5))
+    def test_nonscalar_values_2(self, method, num_trailing_dims):
+        # Verify that non-scalar valued values also work : use different
+        # lengths of axes to simplify tracing the internals
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5),
+                  (0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0, 47)]
+
+        rng = np.random.default_rng(1234)
+
+        trailing_points = [3 + i for i in range(num_trailing_dims)]
+        # NB: values has a `num_trailing_dims` trailing dimension
+        values = rng.random((6, 7, 8, 9, *trailing_points))
+        sample = rng.random(4)   # a single sample point !
+
+        interp = RegularGridInterpolator(points, values, method=method,
+                                         bounds_error=False)
+        v = interp(sample)
+
+        # v has a single sample point *per entry in the trailing dimensions*
+        assert v.shape == (1, *trailing_points)
+
+        # check the values, too : manually loop over the trailing dimensions
+        vs = []
+        for j in range(values.shape[-1]):
+            interp = RegularGridInterpolator(points, values[..., j],
+                                             method=method,
+                                             bounds_error=False)
+            vs.append(interp(sample))
```

Comment:
Aha. I'm going to be annoying and ask to restructure this test. Consider num_trailing_dims=2. Let `values.shape = (6, 7, 8, 9, 2, 3)`, just for the sake of an example. The first four  entries are interpolation axes, and there are 2*3=6 trailing values to broadcast over. I think the test should actually loop over these six values, not only three values it does now. 

The shape of the result is checked above (line 533), so here we only check the values themselves. Options: 1) restrict to the test to two trailing dimensions, type the double loop manually; 2) reshape values to squash the trailing dimensions into one, keep the single for loop, restore the shape of the result before `assert_allclose` in line 548 below --- whichever feels more natural to you.